src/services/setting-service.js

```diff
@@ -5,10 +5,29 @@ class SettingService {
 
   constructor(settingStore) {
     this.settingStore = settingStore;
+    this.defaultSettings = {
+      homeLanguages: [],
+      helpPage: {
+        linkTitle: '',
+        documentNamespace: '',
+        documentSlug: ''
+      },
+      termsPage: {
+        linkTitle: '',
+        documentNamespace: '',
+        documentSlug: ''
+      },
+      footerLinks: [],
+      defaultTags: []
+    };
   }
 
   async getAllSettings() {
     const settings = await this.settingStore.find();
+    if (!settings.length) {
+      return this.defaultSettings;
+    }
```

Comment:
This covers the case when no settings exist at all. 
For consistency I'd suggest to do smth like
```
const settingsArray = await this.settingStore.find();
const settings = settingsArray.reduce((all, { _id, value }) => {
      all[_id] = value;
      return all;
});
return {
   ...defaultSettings,
   ...settings
 }
```